plugins/modules/ipanetgroup.py

```diff
@@ -0,0 +1,369 @@
+    try:
+        _result = module.ipa_command("netgroup_show", name,  {"all": True})
+    except Exception:  # pylint: disable=broad-except
+        # An exception is raised if netgroup name is not found.
+        return None
+    else:
+        return _result["result"]
+
+
+def gen_args(description, nisdomain, nomembers):
+    _args = {}
+    if description is not None:
+        _args["description"] = description
+    if nisdomain is not None:
+        _args["nisdomainname"] = nisdomain
+    if nomembers is not None:
+        _args["nomembers"] = nomembers
+
+    return _args
+
+
+def gen_member_args(user, group, host, hostgroup, netgroup):
+    _args = {}
+    if user is not None:
+        _args["memberuser_user"] = user
+    if group is not None:
+        _args["memberuser_group"] = group
+    if host is not None:
+        _args["memberhost_host"] = host
+    if hostgroup is not None:
+        _args["memberhost_hostgroup"] = hostgroup
+    if netgroup is not None:
+        _args["member_netgroup"] = netgroup
+
+    return _args
+
+
+def main():
+    ansible_module = IPAAnsibleModule(
+        argument_spec=dict(
+            # general
+            name=dict(type="list", aliases=["cn"],
+                      default=None, required=True),
+            # present
+            description=dict(required=False, type='str',
+                             aliases=["desc"], default=None),
+            nisdomain=dict(required=False, type='str',
+                           aliases=["nisdomainname"], default=None),
+            nomembers=dict(required=False, type='bool', default=None),
+            user=dict(required=False, type='list',
+                      aliases=["users"], default=None),
+            group=dict(required=False, type='list',
+                       aliases=["groups"], default=None),
+            host=dict(required=False, type='list',
+                      aliases=["hosts"], default=None),
+            hostgroup=dict(required=False, type='list',
+                           aliases=["hostgroups"], default=None),
+            netgroup=dict(required=False, type='list',
+                          aliases=["netgroups"], default=None),
+            action=dict(type="str", default="netgroup",
+                        choices=["member", "netgroup"]),
+            # state
+            state=dict(type="str", default="present",
+                       choices=["present", "absent"]),
+        ),
+        supports_check_mode=True,
+        ipa_module_options=["delete_continue"],
+    )
+
+    ansible_module._ansible_debug = True
+
+    # Get parameters
+
+    # general
+    names = ansible_module.params_get("name")
+
+    # present
+    description = ansible_module.params_get("description")
+    nisdomain = ansible_module.params_get("nisdomain")
+    nomembers = ansible_module.params_get("nomembers")
+    user = ansible_module.params_get("user")
+    group = ansible_module.params_get("group")
+    host = ansible_module.params_get("host")
```

Comment:
Usually, with `host` members, we have some issues that break module idempotency.

[Check the `hostgroup` module](https://github.com/freeipa/ansible-freeipa/blob/a3a7ecd817f01bebc36ffc83709c1fef5fd93b8a/plugins/modules/ipahostgroup.py#L284-L295), for an example on how to fix.

I'm not sure if similar issues cannot happen with `nisdomain` or `netgroup`.